.gitignore

```diff
@@ -6,4 +6,5 @@
 __pycache__
 build
 venv
-/.idea/
\ No newline at end of file
+/.idea/
```

Comment:
I've left a few comments.

I also think this charm's tests/CI need to be brought up to date a bit. There's an ad-hoc script (`run_tests`) that sets up a virtual environment and runs flake8 and the unit tests, but it looks like that's not run in CI so it'd be easy to forget. We should use tox to manage venvs and run tests. See for example the prometheus-k8s-operator charm's [tox.ini](https://github.com/canonical/prometheus-k8s-operator/blob/main/tox.ini) and [pyproject.toml](https://github.com/canonical/prometheus-k8s-operator/blob/main/pyproject.toml) for an example.

The charm also has very few unit tests. Maybe it's tiny and doesn't need any more, but we should at least add a couple to exercise this new functionality.

Reply:
(Would be worth reading through https://juju.is/docs/sdk/styleguide if you haven't already.)

Reply:
I agree with you re: tox, but it might be out of scope for this PR.

Reply:
Fair enough. However, I'd suggest doing this soon as a follow-up PR. Tox is quite easy to add, and eliminates the need for manually messing with virtual environments and having an ad-hoc `run_tests` script. With my Charm Tech hat on, this charm wouldn't pass review for a real charm as it is ... which doesn't seem like a good thing for a charm used by Juju itself.